internal/kubernetes/controller/manager_builder.go

```diff
@@ -0,0 +1,68 @@
+package controller
+
+import (
+	"github.com/SAP/remote-work-processor/internal/grpc"
+	"github.com/SAP/remote-work-processor/internal/kubernetes/dynamic"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"log"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"time"
+)
+
+type ManagerBuilder struct {
+	delegate      manager.Manager
+	dynamicClient *dynamic.Client
+	grpcClient    *grpc.RemoteWorkProcessorGrpcClient
+}
+
+func NewManagerBuilder() *ManagerBuilder {
+	return &ManagerBuilder{}
+}
+
+func (b *ManagerBuilder) SetGrpcClient(client *grpc.RemoteWorkProcessorGrpcClient) *ManagerBuilder {
+	b.grpcClient = client
+	return b
+}
+
+func (b *ManagerBuilder) BuildDynamicClient(config *rest.Config) *ManagerBuilder {
+	dc, err := dynamic.NewDynamicClient(config)
+	if err != nil {
+		log.Panicln("Failed to create dynamic client:", err)
+	}
+	b.dynamicClient = dc
+	return b
+}
+
+func (b *ManagerBuilder) BuildInternalManager(config *rest.Config, scheme *runtime.Scheme) *ManagerBuilder {
+	t := time.Duration(0)
+	options := manager.Options{
+		Scheme:                  scheme,
+		GracefulShutdownTimeout: &t,
+		WebhookServer:           nil,
+		LeaderElection:          false,
+		HealthProbeBindAddress:  "localhost:8811",
+		MetricsBindAddress:      "0",
+	}
+
+	mgr, err := ctrl.NewManager(config, options)
+	if err != nil {
+		log.Panicln("Failed to create manager:", err)
+	}
+
+	// these can fail only if the manager has been started prior to calling them
+	mgr.AddHealthzCheck("healthz", healthz.Ping)
+	mgr.AddReadyzCheck("readyz", healthz.Ping)
+	b.delegate = mgr
+	return b
+}
```

Comment:
Looking at these, the only reason they are a struct method instead of a separate function is so they can assign the result in the end. Any reason why they are optionals again and are they really optional in our case? Any reason they can't be separate functions that have their results passed as arguments to the actual build function so that we can avoid having optional parameters that are actually always needed?